examples/threading/threading.c

```diff
@@ -14,6 +14,27 @@ void* threadfunc(void* thread_param)
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
     //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+
+
+    struct thread_data* thread_func_args = (struct thread_data*) thread_param;
+    DEBUG_LOG("Starting threadfunc()");
+
+    usleep(thread_func_args->wait_to_obtain_ms * 1000);
+
+    if (pthread_mutex_lock(thread_func_args->mutex) != 0) {
+        perror("mutex lock error");
+        exit(-1);
+    }
+
+    usleep(thread_func_args->wait_to_release_ms * 1000);
+
+    if (pthread_mutex_unlock(thread_func_args->mutex) != 0) {
+        perror("mutex unlock error");
+        exit(-1);
```

Comment:
@bjornhnelson consider whether exit() is the right error strategy here.  I'd use the boolean you have to indicate success/failure of the thread instead of killing the process.